pypeit/slittrace.py

```diff
@@ -0,0 +1,613 @@
+                name, e.g. 'A_1_01'.
+            reuse (:obj:`bool`, optional):
+                Reuse existing master frames from disk. If False,
+                this method returns ``None``.
+
+        Returns:
+            :class:`SlitTraceSet`, None: Returns the instantiated
+            object or ``None``; ``None`` is returned if ``reuse`` is
+            False or the master-frame was not found on disk.
+        """
+        if not reuse:
+            return None
+        # Use MasterFrame to get the name of the file
+        # TODO: This feels like over-kill
+        ifile = masterframe.MasterFrame(cls.master_type, master_dir=master_dir,
+                                        master_key=master_key, file_format=cls.master_format,
+                                        reuse_masters=reuse).chk_load_master(None)
+        if ifile is None:
+            return None
+        msgs.info('Loading SlitTraceSet from: {0}'.format(ifile))
+        return cls.from_file(ifile)
+
+    # NOTE: Don't need to overload the DataContainer.from_file method
+    # because it simply calls this from_hdu method.
+    @classmethod
+    def from_hdu(cls, hdu):
+        """
+        Instantiate the object from an HDU extension.
+
+        This is primarily a wrapper for :func:`_parse`.
+
+        Args:
+            hdu (`astropy.io.fits.HDUList`_, `astropy.io.fits.ImageHDU`_, `astropy.io.fits.BinTableHDU`_):
+                The HDU(s) with the data to use for instantiation.
+        """
+        # NOTE: We can't use `cls(cls._parse(hdu))` here because this
+        # will call the `__init__` method of the derived class and we
+        # need to use the `__init__` of the base class instead. So
+        # below, I get an empty instance of the derived class using
+        # `__new__`, instantiate the MasterFrame components, call the
+        # parent `__init__`, and then return the result. The `__init__`
+        # methods for both base classes are called explicitly to avoid
+        # resolution-order issues with `super`.
+        self = super().__new__(cls)
+        # Instantiate the MasterFrame.
+        hdr = hdu[0].header if isinstance(hdu, fits.HDUList) else hdu.header
+        if hdr['MSTRTYP'] != self.master_type:
+            msgs.warn('Master Type read from header incorrect!  Found {0}; expected {1}'.format(
+                        hdr['MSTRTYP'], self.master_type))
+        # TODO: Throw warning if reuse masters from header is False?
+        masterframe.MasterFrame.__init__(self, self.master_type, master_dir=hdr['MSTRDIR'],
+                                         master_key=hdr['MSTRKEY'], file_format=self.master_format,
+                                         reuse_masters=hdr['MSTRREU'])
+        # Instantiate the DataContainer
+        datamodel.DataContainer.__init__(self, cls._parse(hdu))
+        return self
+
+    # TODO: I think the default overwriting should be False everywhere.
```

Comment:
well,  then we'd have to rely on the user to remember to remove master frames when they restart a run.  Too risky in my opinion..

Reply:
Bottom line for me is that I don't want users to mistakenly overwrite things.

Reply:
I disagree. Clobber should be the default unless specified. Users should be smart enough to make a new directory for their  oldreduction if they want to save it. Using old masters is a big source of traffic in the user channel.

Reply:
I don't understand this, but I doubt I can change your minds with more arguments.  Majority rules.  We should adopt the "overwrite by default" approach everywhere in the code.  I'm sure anything that doesn't do this is my fault, so I'll try to consolidate the code in a branch at some point soon.